pkg/ocm/helpers.go

```diff
@@ -139,13 +139,89 @@ func ValidateHTTPProxy(val interface{}) error {
 		if httpProxy == "" {
 			return nil
 		}
-		url, err := url.ParseRequestURI(httpProxy)
+
+		// Check for spaces
+		if strings.Contains(httpProxy, " ") {
+			return fmt.Errorf("Invalid http-proxy value '%s': URL cannot contain spaces", httpProxy)
+		}
```

Comment:
this gets a little messy when we consider we are asking users to provide URL encoded URLs

This, for example, doesn't check for encoded spaces- and all of this validation becomes mostly useless once they encode

I'm wondering, if it may be a better idea to stick closer to the original ask of the ticket, and maybe not ask users to URL encode specifically. The original ask of the ticket is to tell users what is wrong with the URL they put in; I don't think it mentions telling the user to URL encode if they have an invalid character in their URL. We probably don't want them to URL encode if there's an invalid character, because then we are skipping the CLI and sending the bad value straight to the API